src/components/results/result.js

```diff
@@ -13,14 +13,21 @@ class Result extends React.Component {
 
   render() {
     const {bootstrapCss, doc, fields} = this.props;
+    const Withlink = ({field, link, children})=> (
+        field.field === "id"?
+        <a href={link}>{children}</a> :
+        children
+    );
 
     return (
       <li className={cx({"list-group-item": bootstrapCss})} onClick={() => this.props.onSelect(doc)}>
         <ul>
           {fields.filter((field) => field.field !== "*").map((field, i) =>
             <li key={i}>
               <label>{field.label || field.field}</label>
-              {this.renderValue(field.field, doc)}
+              <Withlink field={field} link={"https://n2t.net/" + this.renderValue(field.field, doc)}>
+                {this.renderValue(field.field, doc)}
+              </Withlink>
```

Comment:
conditionally render solr fields value as a link.

Reply:
I wonder if it might be cleaner to move the conditional down to this layer and make a separate component specific for the link?  I don't know React at all, so I don't necessarily understand the implications here, but it feels like we're mixing concerns a bit.

Reply:
React is new to me so I did some research. I am not sure if this is the best solution since I cannot write a conditional statement inside curly braces with HTML tags. Another solution is to write a conditional statement with two returns which will generate duplicate code. The below link provides the solution of a higher-order component.
https://stackoverflow.com/questions/33710833/how-do-i-conditionally-wrap-a-react-component

Reply:
I don't think duplication is bad if we are doing two different things.  It looks like there are a lot of ways to do conditional rendering in React:

https://reactjs.org/docs/conditional-rendering.html

```
const isId = field.field == "id";
…
…
{ isId
?             <Withlink field={field} link={"https://n2t.net/" + this.renderValue(field.field, doc)}>
                {this.renderValue(field.field, doc)}
              </Withlink>
:            {this.renderValue(field.field, doc)}
}
```

Something like that.

Reply:
This is good. Let me modify it.

Reply:
```
{field.field == "id" ? 
    <Withlink field={field} link={"https://n2t.net/" + this.renderValue(field.field, doc)}>
         {this.renderValue(field.field, doc)}
    </Withlink> 
    : 
     <Withoutlink>
         {this.renderValue(field.field, doc)}
     </Withoutlink>
 }
```
Do you think this one is good? I found we could not use ```{this.renderValue(field.field, doc)}``` directly behind another higher-order wrapper.

Reply:
Hmmm, I don't think we need a `<Withoutlink>` -- that outer wrapper isn't doing anything.  Does this work?

```
{field.field == "id" ? 
    <Withlink field={field} link={"https://n2t.net/" + this.renderValue(field.field, doc)}>
         {this.renderValue(field.field, doc)}
    </Withlink> 
    : 
    this.renderValue(field.field, doc)}
 }
```

Reply:
Nope, it will throw an error. SyntaxError: /Users/qgan/Desktop/iSamples/solr-faceted-search-react/src/components/results/result.js: Unexpected keyword 'this'. (31:19)

  29 |                   </Withlink> 
  30 |                   : 
> 31 |                   {this.renderValue(field.field, doc)}
     |                    ^
  32 |               }
  33 |             </li>
  34 |           )}

Reply:
OK, maybe something like this:

```
  render() {
    const {bootstrapCss, doc, fields} = this.props;
    const Withlink = ({field, link, children})=> (
        <a href={link}>{children}</a> :
    );
    let displayText;
    const isId = field.field == "id";
    if (isId) {
      displayText =  <Withlink field={field} link={"https://n2t.net/" + this.renderValue(field.field, doc)}>{this.renderValue(field.field, doc)}</Withlink>;
    } else {
      displayText = this.renderValue(field.field, doc);
    }

    return (
      <li className={cx({"list-group-item": bootstrapCss})} onClick={() => this.props.onSelect(doc)}>
        <ul>
          {fields.filter((field) => field.field !== "*").map((field, i) =>
            <li key={i}>
              <label>{field.label || field.field}</label>
              {displayText}
            </li>
          )}
        </ul>
      </li>
    );
  }
```

Reply:
If that isn't exactly right, please consult the react reference -- it's definitely possible to conditionally render components and it's the right thing to do in this case.

Reply:
This one works without other components.
```
    return (
      <li className={cx({"list-group-item": bootstrapCss})} onClick={() => this.props.onSelect(doc)}>
        <ul>
          {fields.filter((field) => field.field !== "*").map((field, i) =>
            <li key={i}>
              <label>{field.label || field.field}</label>
              {field.field === "id"?
                <a href={"https://n2t.net/" + this.renderValue(field.field, doc)}>{this.renderValue(field.field, doc)}</a>
                :
                this.renderValue(field.field, doc)
              }
            </li>
          )}
        </ul>
      </li>
    );
```

Reply:
That looks simple and clean to me.  Thanks!